core/src/SYCL/Kokkos_SYCL_Space.hpp

```diff
@@ -301,10 +301,11 @@ struct MemorySpaceAccess<Kokkos::Experimental::SYCLHostUSMSpace,
   enum : bool { deepcopy = true };
 };
 
-template <>
+template <typename LocalPointerType, typename GlobalPointerType>
 struct MemorySpaceAccess<
     Kokkos::Experimental::SYCLDeviceUSMSpace,
-    Kokkos::ScratchMemorySpace<Kokkos::Experimental::SYCL>> {
+    Kokkos::ScratchMemorySpace<Kokkos::Experimental::SYCL, LocalPointerType,
+                               GlobalPointerType>> {
```

Comment:
Without this fix, I was getting segmentation faults when the memory space availability check triggered instead of a message through `Kokkos::abort`.